app/src/main/java/org/fossasia/openevent/general/StartupViewModel.kt

```diff
@@ -116,4 +120,23 @@ class StartupViewModel(
                 Timber.e(it, "Error in fetching settings form API")
             })
     }
+
+    fun checkAccessToken() {
+        if (!authHolder.isLoggedIn() || authHolder.isTokenValid())
+            return
+        preference.putString(HEADER_TYPE, JWT_REFRESH_TOKEN)
+        compositeDisposable += authService.refreshToken()
+            .withDefaultSchedulers()
+            .doFinally {
+                preference.putString(HEADER_TYPE, JWT_ACCESS_TOKEN)
+            }.subscribe({
+                authHolder.accessToken = it.accessToken
```

Comment:
so header type is still set to JWT_ACCESS_TOKEN even though `subscribe()` throws an error?

Reply:
Yes, it will logout the user due to the unauthorization error.

Reply:
Add we are using access token for all API calls